src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/WorkflowRunController.kt

```diff
@@ -129,11 +138,21 @@ class WorkflowRunController(
             .get("/v1/workflow/$key/status/", emptyMap())
         val workflowRunStatusResponse = response.data(WorkflowRunStatusResponse::class.java)
         workflowRunRepository.updateWorkflowRun(key, workflowRunStatusResponse.status)
-        val reportNames = workflowRunRepository.getWorkflowRunDetails(key).reports.map { it.name }
+
         return WorkflowRunStatus(
             workflowRunStatusResponse.status,
-            workflowRunStatusResponse.reports.zip(reportNames) { report, reportName ->
-                WorkflowRunStatus.WorkflowRunReportStatus(reportName, report.key, report.status, report.version)
+            workflowRunStatusResponse.reports.map { report ->
+                WorkflowRunStatus.WorkflowRunReportStatus(
+                    getReportName(report.key),
+                    report.key,
+                    report.status,
+                    report.version
+                )
             })
     }
+
+    private fun getReportName(key: String): String
+    {
+        return workflowRunRepository.getWorkflowRunReportsByReportKey(key).name
```

Comment:
I thought we had agreed in the Teams conversation that `getWorkflowRunReportsByReportKey` should be removed from the repository, and that `getWorkflowRunStatus` would instead call `getWorkflowRunDetails` and manage matching up the keys from the details with the keys returned from the orderly server status endpoint.

You could keep this private method, and pass it the `WorkflowRun` returned by `getWorkflowRunDetails`, and do something like `return workflowRun.workflowRunReports.find{it.reportKey == key}.name`
(although, see my comment on the model class, I think `workflowRun.workflowRunReports`can just become `workflowRun.reports`) - or any other implementation you prefer.

Reply:
I agree with this (as well as the rest of Emma's comments). It remains my view that `getWorkflowRunDetails()`, as amended, is sufficient - and should be used by this class in preference to `getWorkflowRunReportsByReportKey()` which I think unnecessarily expands the `WorkflowRunResponsibility` interface (and duplicates some logic therein).